trieste/acquisition/function/active_learning.py

```diff
@@ -245,3 +245,153 @@ def acquisition(x: TensorType) -> TensorType:
+        :param threshold: either None, or a sequence of 1 or 2 float values
+        :raise ValueError (or InvalidArgumentError): If ``threshold`` has more than 2 values.
+        """
+        if not hasattr(model, "conditional_predict_f"):
+            raise AttributeError(
+                """
+                Integrated variance reduction only supports models with a
+                conditional_predict_f method.
+                """
+            )
+
+        self._model = model
+
+        tf.debugging.assert_equal(
+            len(tf.shape(integration_points)),
+            2,
+            message="integration_points must be of shape [N, D]",
+        )
+
+        tf.debugging.assert_positive(
+            tf.shape(integration_points)[0],
+            message="integration_points should contain at least one point",
+        )
+
+        self._integration_points = integration_points
+
+        if threshold is not None:
+            threshold = tf.cast(threshold, integration_points.dtype)
+
+            tf.debugging.assert_rank(
+                threshold,
+                1,
+                message=f"threshold should be a sequence "
+                f"or a rank 1 tensor, received {tf.shape(threshold)}",
+            )
+            tf.debugging.assert_less_equal(
+                tf.size(threshold),
+                2,
+                message=f"threshold should have one or two values,"
+                f" received {tf.size(threshold)}",
+            )
+            tf.debugging.assert_greater_equal(
+                tf.size(threshold),
+                1,
+                message=f"threshold should have one or two values,"
+                f" received {tf.size(threshold)}",
+            )
+
+        if threshold is None:
+            self._weights = tf.cast(1.0, integration_points.dtype)
+        elif len(threshold) == 1:
+            mean_old, var_old = self._model.predict(query_points=integration_points)
+            distr = tfp.distributions.Normal(mean_old, tf.sqrt(var_old))
+            self._weights = distr.prob(threshold)
+        else:
+            mean_old, var_old = self._model.predict(query_points=integration_points)
+            distr = tfp.distributions.Normal(mean_old, tf.sqrt(var_old))
+            self._weights = distr.cdf(threshold[1]) - distr.cdf(threshold[0])
+
+    @tf.function
+    def __call__(self, x: TensorType) -> TensorType:
+
+        additional_data = Dataset(x, tf.ones_like(x[..., 0:1]))
+
+        try:
+            mean, variance = self._model.conditional_predict_f(  # type: ignore
+                query_points=self._integration_points, additional_data=additional_data
+            )
+        except NotImplementedError:
+            raise ValueError(
+                """
+                integrated_variance_reduction only supports models with a conditional_predict_f
+                method.
+                """
+            )
+
+        return -tf.reduce_mean(variance * self._weights, axis=-2)
```

Comment:
why minus here? since timse is maximised same as Bichon, we shouldn't have minus, optimizer adds a minus

Reply:
actually the criterion to maximise is the variance **reduction**, so it should write   
`tf.reduce_mean((old_variance - new_variance) * self._weights, axis=-2)`

but since `old_variance` just adds a constant term and incurs extra cost, I don't use it here. I'll explain this in the docstring.